core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -528,7 +528,13 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
         new StageUIData
       })
       val taskData = stageData.taskData.get(taskId)
-      val metrics = TaskMetrics.fromAccumulatorInfos(accumUpdates)
+      val accumsFiltered = if (conf.get(TASK_METRICS_TRACK_UPDATED_BLOCK_STATUSES)) {
+        accumUpdates
+      } else {
+        accumUpdates.filter(info => info.name.isDefined && info.update.isDefined && info.name !=
```

Comment:
yes that is correct as I said it shouldn't really ever get there so I can add an assert.  Sorry for the delay was out of office for a while. I'll update it.